include/nuttx/fs/fs.h

```diff
@@ -490,7 +490,6 @@ struct file
 
 struct filelist
 {
-  spinlock_t        fl_lock;    /* Manage access to the file list */
```

Comment:
How removal of `fl_lock` is related to `inline spin_lock`? I do not see any mentioning of `fl_lock` removal in commit message

Reply:
We are doing this to remove the dependency of fs.h on spinlock.h, because the inclusion of <atomic> introduces a dependency on fs.h, which in turn creates a circular dependency between spinlock.h and fs.h.

Reply:
@hujun260 @xiaoxiang781216 So do we have a plan on when to revert this change? Is it just going to stay that way forever?

Reply:
We have no plans to revert this patch.

Reply:
we will switching back to mutex, it's wrong to use spinlock here.

Reply:
replace to a mutex lock may regress to previous performance
https://github.com/apache/nuttx/pull/11140#issuecomment-1803541935

Reply:
> We have no plans to revert this patch.

I mean not revert inlin spinlock, just revert fl_lock to use local spinlock

Reply:
The right approach is to optimize mutex performance. @zyfeier will optimize mutex implementation to get the same performance as spinlock when no contention happens, after that the code will restore to mutex.